src/ServiceControl/Monitoring/MonitorEndpointsFoundInFailedMessageHeaders.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace ServiceControl.EndpointControl.Handlers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Monitoring;
+    using NServiceBus.Transport;
+    using Operations;
+    using ServiceControl.Contracts.Operations;
+
+    class MonitorEndpointsFoundInFailedMessageHeaders : IErrorMessageBatchPlugin
+    {
+        public Task AfterProcessing(List<MessageContext> batch)
+        {
+            var observedEndpoints = new HashSet<EndpointDetails>();
```

Comment:
Would it be better to have a singleton comparer that uses the unique key of the details instead of relying on the default comparer?